### ISSUE TITLE
Dose every dataset need to transform the images to [-0.5 0.5]?

### ISSUE THREAD

**Issue by Ironbrotherstyle** [ClementPinard/FlowNetPytorch#58]
Hi, I noticed that in your readme wrote: 
**Note on networks from caffe**
These networks expect a BGR input in range [-0.5,0.5] (compared to RGB in pytorch). However, BGR order is not very important.

Does this mean that I need to set all my input images to be between [-0.5 0.5] on pytorch when using your model?

**Comment by ClementPinard** [comment]
If you don't do retraining, yes you need to normalize images that way. If you apply a fine-tuning, it's probably better to do it but it's less of a problem.

**Comment by Ironbrotherstyle** [comment]
> If you don't do retraining, yes you need to normalize images that way. If you apply a fine-tuning, it's probably better to do it but it's less of a problem.
I want to use your model as pre-trained model to help to converge on KITTI dataset, so may update the weights in Flownet, should I set all my input images to be between [-0.5 0.5]? Thank you so much!

**Comment by ClementPinard** [comment]
You already have a training for KITTI dataset within this repo, and it normalizes images.

Also the pytorch pretrained networks now also expects shifted images to have values inside [-0.5,0.5], so by just using standard data transform module used in the main.py, you should be good.


Other than that, the thing is there is no specific thing you "should do", I am not in your project, I don't really know in what context you actually work on. You don't need to tell it for me to give better advice, the idea is that it may change for every project, so maybe there is a thumb rule of normalizing images, but I can't say for sure it will be best for you.

What if you worked with very dark images ? What if you your images have already a very high contrast ?
This network theoretically works for every kind of luminosity distribution, but it has been trained on 0-centered images. As such it will be better on them without finetuning, but if your images have something else, finetuning might work for different distributions.

All in all, it's up to you, to decide to normalize images or not for finetuning, and I can't know it for you.

**Comment by Ironbrotherstyle** [comment]
> You already have a training for KITTI dataset within this repo, and it normalizes images.
> 
> Also the pytorch pretrained networks now also expects shifted images to have values inside [-0.5,0.5], so by just using standard data transform module used in the main.py, you should be good.
> 
> Other than that, the thing is there is no specific thing you "should do", I am not in your project, I don't really know in what context you actually work on. You don't need to tell it for me to give better advice, the idea is that it may change for every project, so maybe there is a thumb rule of normalizing images, but I can't say for sure it will be best for you.
> 
> What if you worked with very dark images ? What if you your images have already a very high contrast ?
> This network theoretically works for every kind of luminosity distribution, but it has been trained on 0-centered images. As such it will be better on them without finetuning, but if your images have something else, finetuning might work for different distributions.
> 
> All in all, it's up to you, to decide to normalize images or not for finetuning, and I can't know it for you.

Very nice of you, thank you!
